components/Dashboard/Settings/SocialForm.tsx

```diff
@@ -3,81 +3,118 @@ import { useForm } from 'react-hook-form'
 import { useTranslation } from '../../../config/i18n'
 import SettingsForm from '../../../components/Dashboard/Settings/SettingsForm'
 import SettingsFieldset from '../../../components/Dashboard/Settings/SettingsFieldset'
-import XIcon from '../../../components/Icons/XIcon'
 import Button, { ButtonVariant } from '../../../elements/Button'
 import theme from '../../../theme'
+import { SocialPlatform, SocialFormField } from './SocialFormField'
+import { toLower, upperCase } from 'lodash'
+import FormError from '../../FormError'
+import { SocialMedia, useUpdateSocialMediaMutation } from '../../../generated/graphql'
+import { toast } from 'react-toastify'
+import { ApolloError } from '@apollo/client'
 
-const SocialForm: React.FC = () => {
+type FormData = {
+  facebook: string
+  instagram: string
+  youtube: string
+  website: string
+  [key: string]: string
+}
+
+const getPlatformValue = (platform: string, data: SocialMedia[]): string => {
+  const socialMedia = data?.find(
+    (socialMedia) => upperCase(socialMedia.platform) === upperCase(platform),
+  )
+  return socialMedia?.url || ''
+}
+
+type SocialFormProps = {
+  socialMedia: SocialMedia[]
+  refetch: () => void
+}
+
+const SocialForm: React.FC<SocialFormProps> = ({ socialMedia, refetch }) => {
   const { t } = useTranslation('settings')
-  const { handleSubmit, register } = useForm({
+  const { handleSubmit, register, errors, setError } = useForm<FormData>({
     mode: 'onSubmit',
     reValidateMode: 'onBlur',
   })
 
-  const handleSocialSubmit = (): void => {}
+  const [updateSocialMedia, { loading: updateInProgress }] = useUpdateSocialMediaMutation({
+    onCompleted: () => {
+      toast.success(t('profile.social.success'))
+    },
+  })
+
+  const addServerErrors = (apolloError: ApolloError): void => {
+    const badRequest = (apolloError.networkError as any)?.result?.errors[0].extensions
+    if (!badRequest || badRequest.statusCode !== 400) {
+      toast.error(t('profile.error.updateError'))
+      return
+    }
+
+    badRequest.validationErrors.forEach((err: any): void => {
+      setError(err.name, 'server', t(err.message))
+    })
+  }
+
+  const handleSocialSubmit = async (formData: FormData): Promise<void> => {
+    if (formData) {
+      await updateSocialMedia({
+        variables: {
+          facebook: formData.facebook,
+          instagram: formData.instagram,
+          youtube: formData.youtube,
+          website: formData.website,
+        },
+      })
+        .then(() => {
+          refetch()
+        })
+        .catch((err) => {
+          addServerErrors(err)
+        })
+    }
+  }
```

Comment:
If we're in an `async` function we don't use typically use `then`/`catch`. Instead use `try/catch`